pkg/governance-scripts/test/TimelockAuthorizerTransitionMigrator.test.ts

```diff
@@ -0,0 +1,151 @@
+  const ROLE_2 = '0x0000000000000000000000000000000000000000000000000000000000000002';
+  const ROLE_3 = '0x0000000000000000000000000000000000000000000000000000000000000003';
+
+  sharedBeforeEach('set up vault', async () => {
+    oldAuthorizer = await deploy('v2-vault/MockBasicAuthorizer', { from: oldRoot });
+    vault = await deploy('v2-vault/Vault', { args: [oldAuthorizer.address, ZERO_ADDRESS, 0, 0] });
+
+    const authorizerAdaptor = await deploy('v2-liquidity-mining/AuthorizerAdaptor', { args: [vault.address] });
+    adaptorEntrypoint = await deploy('v2-liquidity-mining/AuthorizerAdaptorEntrypoint', {
+      args: [authorizerAdaptor.address],
+    });
+  });
+
+  sharedBeforeEach('set up permissions', async () => {
+    const target = await deploy('v2-vault/MockAuthenticatedContract', { args: [vault.address] });
+    rolesData = [
+      { grantee: user1.address, role: ROLE_1, target: target.address },
+      { grantee: user2.address, role: ROLE_2, target: target.address },
+      { grantee: user3.address, role: ROLE_3, target: ZERO_ADDRESS },
+    ];
+  });
+
+  sharedBeforeEach('grant roles on old Authorizer', async () => {
+    await oldAuthorizer
+      .connect(oldRoot)
+      .grantRolesToMany([ROLE_1, ROLE_2, ROLE_3], [user1.address, user2.address, user3.address]);
+  });
+
+  sharedBeforeEach('deploy new authorizer', async () => {
+    newAuthorizer = await deploy('TimelockAuthorizer', { args: [root.address, adaptorEntrypoint.address, 0] });
+  });
+
+  sharedBeforeEach('deploy migrator', async () => {
+    const args = [oldAuthorizer.address, newAuthorizer.address, rolesData];
+    transitionMigrator = await deploy('TimelockAuthorizerTransitionMigrator', { args });
+  });
+
+  describe('constructor', () => {
+    context('when attempting to migrate a role which does not exist on previous Authorizer', () => {
+      let tempAuthorizer: Contract;
+
+      sharedBeforeEach('set up vault', async () => {
+        tempAuthorizer = await deploy('v2-vault/MockBasicAuthorizer');
+      });
+
+      it('reverts', async () => {
+        const args = [tempAuthorizer.address, newAuthorizer.address, rolesData];
+        await expect(deploy('TimelockAuthorizerTransitionMigrator', { args })).to.be.revertedWith('UNEXPECTED_ROLE');
+      });
+    });
+  });
+
+  describe('migrate permissions', () => {
+    context('when the migrator is not a granter', () => {
+      it('reverts', async () => {
+        await expect(transitionMigrator.migratePermissions()).to.be.revertedWith('SENDER_NOT_ALLOWED');
+      });
+    });
+
+    context('when the migrator is a granter', () => {
+      sharedBeforeEach(async () => {
+        await newAuthorizer
+          .connect(root)
+          .manageGranter(
+            newAuthorizer.GENERAL_PERMISSION_SPECIFIER(),
+            transitionMigrator.address,
+            newAuthorizer.EVERYWHERE(),
+            true
+          );
+      });
+
+      it('migrates all roles properly', async () => {
```

Comment:
I tried this change but then I stepped back since we already have an outer `describe` for that. The current version, even though more redundant, is also more readable and follows the style of other tests (i.e. performing the operation in every `it` block).